flocker/node/agents/blockdevice.py

```diff
@@ -779,6 +779,20 @@ def _manifestation_from_volume(volume):
     )
     return Manifestation(dataset=dataset, primary=True)
 
+# FLOC-1513
+#
+# Introduce a new PRecord, ``VisibleClusterState``, with a ``NodeState`` field
+# and a ``nonmanifest_datasets`` field.
+#
+# It also has a method, update_cluster_state, which both updates the cluster
+# with its ``NodeState`` (just delegate to that object's update_cluster_state
+# method) and with its own ``nonmanifest_datasets`` (use
+# ``DeploymentState.update_nonmanifest_datasets``).
+#
+# Probably introduce an interface for this thing now.  It's what must be
+# returned by ``discover_local_state`` and it must have this
+# ``update_cluster_state`` method.
+
```

Comment:
Replace with `NonManifestDatasets` class which implements `IClusterStateUpdater`. (which may as well reside in `flocker.control._model`) Don't forget to add it to `_model.SERIALIZABLE_CLASSES`.